lms/djangoapps/course_api/blocks/forms.py

```diff
@@ -9,37 +9,28 @@
 
 from opaque_keys import InvalidKeyError
 from opaque_keys.edx.keys import UsageKey
-from openedx.core.djangoapps.util.forms import MultiValueField
+from openedx.core.djangoapps.util.forms import ExtendedNullBooleanField, MultiValueField
 from xmodule.modulestore.django import modulestore
 
-from .permissions import can_access_other_users_blocks, can_access_users_blocks
+from . import permissions
 
 
 class BlockListGetForm(Form):
     """
     A form to validate query parameters in the block list retrieval endpoint
     """
-    username = CharField(required=True)  # TODO return all blocks if user is not specified by requesting staff user
-    usage_key = CharField(required=True)
-    requested_fields = MultiValueField(required=False)
-    student_view_data = MultiValueField(required=False)
+    all_blocks = ExtendedNullBooleanField(required=False)
     block_counts = MultiValueField(required=False)
     depth = CharField(required=False)
     nav_depth = IntegerField(required=False, min_value=0)
+    requested_fields = MultiValueField(required=False)
     return_type = ChoiceField(
         required=False,
         choices=[(choice, choice) for choice in ['dict', 'list']],
     )
-
-    def clean_requested_fields(self):
-        """
-        Return a set of `requested_fields`, merged with defaults of `type`
-        and `display_name`
-        """
-        requested_fields = self.cleaned_data['requested_fields']
-
-        # add default requested_fields
-        return (requested_fields or set()) | {'type', 'display_name'}
+    student_view_data = MultiValueField(required=False)
+    usage_key = CharField(required=True)
+    username = CharField(required=False)
```

Comment:
Note to reviewers: The pre-existing list of fields and clean_X methods were only reordered alphabetically for readability.  Only the implementation of the `clean_requested_user` method was changed.